### DISCUSSION TITLE
Unable to log

### DISCUSSION THREAD

**Discussion by howell-snc** [inspircd/inspircd#2023] [Support]
**Description**

Unable to enable logging

**Steps to reproduce the issue:**

1. Setup logging in the config file
```
<path logdir="logs">

<log method="file"
     type="*"
     level="debug"
     target="ircd.log"
     flush="100">
```

2.  create a logs folder for logging
mkdir /etc/inspircd/logs && chown irc:irc /etc/inspircd/logs
3. start the service
systemctl restart inspircd
4. Check for logs under /etc/inspircd/logs

I would expect to see something similar to the output of `journalctl -u inspircd.service` logged to /etc/inspircd/logs/ircd.log or even under /var/logs/inspircd.log.

I see no ircd.log file created under /etc/inspircd/logs.

I see there was a similar issue here: https://github.com/inspircd/inspircd-docker/issues/106
But I am using the standard file name and still experience this issue.

`./bin/inspircd --version`: InspIRCd-3.12.0-debian

**Comment by genius3000** [comment]
I believe the Debian package would be `/var/log/inspircd` by default. Your declaration of a different `logdir` might actually be trying to use `/logs` - "logs" being relative to a base directory of '/'.
You may be silently encountering errors of 'no such directory' or permissions. AppArmor might be blocking something unless you've modified it to allow the different paths.

Also to note, `flush="100"` will buffer log lines to 100 before writing to disk; I don't recall if it will create a missing file prior to or not.

**Comment by sadiepowell** [comment]
@genius3000 is correct about AppArmor and the default path although `<paths>` is also not relative to anything so you need to specify an absolute path.

Also, 100 is an incredibly high flush value for logging. You should probably lower it to something sensible like 10 or omit it to use the default.

You may also want to consider using our .deb packages as they're more frequently updated than the Debian ones.

**Comment by howell-snc** [comment]
Thank you both for the suggestions. Things are working as expected now using the .deb package and this configuration:
```
<path logdir="/etc/inspircd/logs">

<log method="file"
     type="*"
     level="debug"
     target="ircd.log"
     flush="10">
```

Would it be possible to update the documentation to specify the path is absolute? https://docs.inspircd.org/3/configuration/#path
The current example in the documentation lead me to believe it was a relative path to /etc/inspircd

**Reply by sadiepowell** [reply]
I've updated that page to make it more clear.
